stock_picking_mto/models/stock_picking.py

```diff
@@ -0,0 +1,40 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def _get_selection_procure_method(self):
+        return self.env['stock.move'].fields_get(
+            allfields=['procure_method'])['procure_method']['selection']
+
+    def _compute_procure_method(self):
+        for picking in self:
+            procure_method = False
+            for move in picking.move_lines:
+                if not procure_method:
+                    procure_method = move.procure_method
+                elif procure_method != move.procure_method:
+                    procure_method = False
+                break
+            picking.procure_method = procure_method
+
+    def _inverse_procure_method(self):
+        self.filtered('procure_method').mapped('move_lines').write({
```

Comment:
@chienandalu Never do a write in an inverse. This will trigger a new recomputation. Simply affect recordset fields.

Reply:
Do `update` instead of `write`